### PR TITLE
taxonomies-refactor

### PR DESCRIPTION
**Pull request by ramirobg94**
This is the base branch, 

the first commit is about moving the files and rename some things following our conventions.

### CODE REVIEW

File: webapp/components/survey/NodeDefDetails/TaxonProps.js [hunk 1]
```diff
@@ -98,7 +98,7 @@ const TaxonProps = (props) => {
             }}
             header={i18n.t('taxonomy.header')}
           >
-            <TaxonomyView showClose={false} />
+            <TaxonomyComponent showClose={false} />
```
Comment (ramirobg94) [inline webapp/components/survey/NodeDefDetails/TaxonProps.js:101]: I put TaxonomyComponnet due to we are using the Taxonomy from Core.

Comment (minotogna) [reply webapp/components/survey/NodeDefDetails/TaxonProps.js:101]: maybe call this TaxonomyDetails . more consistent with NodeDefDetails as well

File: webapp/components/survey/NodeDefDetails/TaxonProps.js [hunk 2]
```diff
@@ -11,15 +11,15 @@ import * as Validation from '@core/validation/validation'
...
-import TaxonomyView from '@webapp/loggedin/surveyViews/taxonomy/taxonomyView'
 
 import { useI18n } from '@webapp/store/system'
 import { useSurvey } from '@webapp/store/survey'
 
-import * as TaxonomyActions from '@webapp/loggedin/surveyViews/taxonomy/actions'
-import * as TaxonomyState from '@webapp/loggedin/surveyViews/taxonomy/taxonomyState'
-
 import TaxonomyList from '@webapp/views/App/views/Designer/TaxonomyList'
+import TaxonomyComponent from '@webapp/views/App/views/Designer/Taxonomy'
```
Comment (minotogna) [inline webapp/components/survey/NodeDefDetails/TaxonProps.js:19]: they should both go under survey common components

File: webapp/views/App/views/Designer/Designer.js [hunk 3]
```diff
@@ -16,6 +15,7 @@ import { resetForm } from '@webapp/loggedin/surveyViews/surveyForm/actions'
 import FormDesigner from './FormDesigner'
 import SurveyHierarchy from './SurveyHierarchy'
 import TaxonomyList from './TaxonomyList'
+import Taxonomy from './Taxonomy'
```
Comment (minotogna) [inline webapp/views/App/views/Designer/Designer.js:18]: call this TaxonomyDetails

File: webapp/components/survey/NodeDefDetails/TaxonProps.js [hunk 4]
```diff
@@ -89,25 +80,31 @@ const TaxonProps = (props) => {
...
               setShowTaxonomyPanel(false)
             }}
             header={i18n.t('taxonomy.header')}
           >
-            <TaxonomyView showClose={false} />
+            <TaxonomyDetails showClose={false} onTaxonomyCreated={onTaxonomySelect} />
           </PanelRight>
         )}
-        {!editedTaxonomy && showTaxonomiesPanel && (
+        {!showTaxonomyPanel && showTaxonomiesPanel && (
```
Comment (minotogna) [inline webapp/components/survey/NodeDefDetails/TaxonProps.js:94]: why not to have just one variable panel which can be null || taxonomy || taxonomies ?

File: webapp/components/survey/TaxonomyDetails/EditHeader/EditHeader.js [hunk 5]
```diff
@@ -15,15 +15,14 @@ import { useI18n } from '@webapp/store/system'
 import { useSurveyId } from '@webapp/store/survey'
 import { useAuthCanEditSurvey } from '@webapp/store/user'
 
-import * as TaxonomyActions from '../actions'
-import * as TaxonomyState from '../taxonomyState'
+import { State } from '../store'
 
-const TaxonomyEditHeader = () => {
-  const dispatch = useDispatch()
+const EditHeader = (props) => {
```
Comment (minotogna) [inline webapp/components/survey/TaxonomyDetails/EditHeader/EditHeader.js:20]: why not just header? it can be readOnly for some user

Comment (ramirobg94) [reply webapp/components/survey/TaxonomyDetails/EditHeader/EditHeader.js:20]: Agree!

File: webapp/components/survey/NodeDefDetails/TaxonProps.js (hunk 4 above)
Comment (ramirobg94) [reply webapp/components/survey/NodeDefDetails/TaxonProps.js:94]: i will think tomorrow about that

File: webapp/components/survey/TaxonomyDetails/actions.js [hunk 6]
```diff
@@ -0,0 +1,8 @@
+// Taxonomy editor actions
+export const taxonomyViewTaxonomyUpdate = 'taxonomyView/taxonomy/update'
+export const taxonomyViewTaxonomyPropsUpdate = 'taxonomyView/taxonomy/props/update'
+
+// ====== SET TAXONOMY FOR EDIT
+
+export const setTaxonomyForEdit = (taxonomyUuid) => (dispatch) =>
```
Comment (minotogna) [inline webapp/components/survey/TaxonomyDetails/actions.js:7]: remove taxonomy redux store

File: webapp/components/survey/TaxonomyDetails/TaxonomyDetails.scss [hunk 7]
```diff
@@ -5,7 +5,10 @@
   grid-template-rows: 4rem 1fr 3rem;
   align-items: center;
   height: 100%;
+}
 
+.taxonomy__table-container {
+  height: 100%;
```
Comment (minotogna) [inline webapp/components/survey/TaxonomyDetails/TaxonomyDetails.scss:11]: something fishy here :P was that needed?

File: webapp/components/survey/TaxonomyList/Row/store/actions/useDelete.js [hunk 8]
```diff
@@ -28,8 +29,14 @@ export const useDelete = () => {
           DialogConfirmActions.showDialogConfirm({
             key: 'taxonomy.confirmDelete',
             params: { taxonomyName: Taxonomy.getName(taxonomy) || i18n.t('common.undefinedName') },
-            onOk: () => {
-              dispatch(TaxonomyActions.deleteTaxonomy(taxonomy, State.getInitData(state)))
+            onOk: async () => {
+              dispatch({ type: TaxonomiesActions.taxonomyDelete, taxonomy })
```
Comment (minotogna) [inline webapp/components/survey/TaxonomyList/Row/store/actions/useDelete.js:33]: you might not need this

File: webapp/components/survey/TaxonomyList/Row/store/actions/useDelete.js [hunk 9]
```diff
@@ -28,8 +29,14 @@ export const useDelete = () => {
...
+              const { data } = await axios.delete(`/api/survey/${surveyId}/taxonomies/${Taxonomy.getUuid(taxonomy)}`)
+
+              dispatch({ type: TaxonomiesActions.taxonomiesUpdate, taxonomies: data.taxonomies })
```
Comment (minotogna) [inline webapp/components/survey/TaxonomyList/Row/store/actions/useDelete.js:36]: not even this

File: webapp/components/survey/TaxonomyList/Row/store/actions/useEdit.js [hunk 10]
```diff
@@ -1,28 +1,31 @@
...
+  const dispatch = useDispatch()
   const { pathname } = useLocation()
   const inTaxonomiesPath = Boolean(matchPath(pathname, appModuleUri(designerModules.taxonomies)))
 
   return useCallback(({ state }) => {
     const taxonomy = State.getTaxonomy(state)
+    const taxonomyUuid = Taxonomy.getUuid(taxonomy)
     if (!inTaxonomiesPath) {
-      dispatch(TaxonomyActions.setTaxonomyForEdit(Taxonomy.getUuid(taxonomy)))
+      dispatch(TaxonomyActions.setTaxonomyForEdit(taxonomyUuid))
```
Comment (minotogna) [inline webapp/components/survey/TaxonomyList/Row/store/actions/useEdit.js:22]: you dont need this

File: webapp/loggedin/surveyViews/reducer.js [hunk 11]
```diff
@@ -1,14 +1,15 @@
 import { combineReducers } from 'redux'
 
+import * as TaxonomyState from '@webapp/components/survey/TaxonomyDetails/taxonomyState'
+import taxonomyEdit from '@webapp/components/survey/TaxonomyDetails/reducer'
```
Comment (minotogna) [inline webapp/loggedin/surveyViews/reducer.js:4]: this should go away

File: webapp/components/survey/TaxonomyList/Row/store/actions/useDelete.js (hunk 8 above)
Comment (ramirobg94) [reply webapp/components/survey/TaxonomyList/Row/store/actions/useDelete.js:33]: do we need to put the survey dirty?

File: webapp/components/survey/TaxonomyDetails/TaxonomyDetails.scss (hunk 7 above)
Comment (ramirobg94) [reply webapp/components/survey/TaxonomyDetails/TaxonomyDetails.scss:11]: Captura de pantalla 2020-07-10 a las 13 00 33
:/ i think that yes

Comment (ramirobg94) [reply webapp/components/survey/TaxonomyDetails/TaxonomyDetails.scss:11]: maybe as we use grid we can apply `align-self: stretch;`what do you think?

**Pull request merged**